### PR TITLE
Setup documentation

### LINKED ISSUE #41: Code and usage documentation
Generate code and usage documentation to make contributing and using m4e easier. 
I suggest we use the following:
- Generate the documentation: [sphix](https://www.sphinx-doc.org/en/master/)
- Hosting of documentation: [readthedocs](https://readthedocs.org)
- Docstring style: Either numpy or google style (comparison [here](https://stackoverflow.com/a/24385103/9356180))
- Avoid GitHub Wiki (readthedocs supports selecting specific documentation versions / releases)

Differentiation from README:
@bjrne issue #33 describes an overview page and how to get started (demo, installing m4e)
The readthedocs documentation describes how to set up m4e and how the code works

### PR DESCRIPTION
**Pull request by Baschdl**
Now we just need to write it

Closes #41

### CODE REVIEW

**Comment by bjrne** [comment]
> Now we just need to write it

Shouldn't we include the index file here already? Since rtd already links to it:
Screenshot from 2020-07-09 19-16-20_cropped
You probably know how to generate one and I would need to google it before getting started on writing :)

Will test the generation locally.

**Comment by Baschdl** [comment]
It's already included: https://github.com/match4everyone/match4everything/blob/d69303e802bf41f0b5b0ff05ca3d3a605227d193/docs/index.rst

We just have to merge this PR

File: docs/make.bat [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	echo.
+	echo.The 'sphinx-build' command was not found. Make sure you have Sphinx
+	echo.installed, then set the SPHINXBUILD environment variable to point
+	echo.to the full path of the 'sphinx-build' executable. Alternatively you
+	echo.may add the Sphinx directory to PATH.
+	echo.
+	echo.If you don't have Sphinx installed, grab it from
+	echo.http://sphinx-doc.org/
+	exit /b 1
+)
```
Comment (bjrne) [inline docs/make.bat:26]: I did not get this error. I got this:
```bash
/bin/sh: 1: sphinx-build: not found
make: *** [Makefile:20: html] Error 127
```
So does this script work? How would I execute it?

**Review by bjrne — CHANGES_REQUESTED** [review]
This needs more documentation or the script needs to be adjusted.

File: docs/make.bat (hunk 1 above)
Comment (bjrne) [reply docs/make.bat:26]: With the requirements it works out of the box

**Review by bjrne — APPROVED** [review]

**Pull request merged**